omas/omas_physics.py

```diff
@@ -387,24 +387,31 @@ def summary_taue(ods, update=True):
             isotope_factor = (2.014102 * n_deuterium_avg + 3.016049 * n_tritium_avg) / (n_deuterium_avg + n_tritium_avg)
 
             # Get total power from ods function:
-            ods.physics_summary_total_powers()
-            total_power = ods['summary.global_quantities.power_steady.value'][time_index]
+            if 'core_sources' not in ods:
+                if 'summary.global_quantities.power_steady.value' not in ods:
+                    raise Exception('No core sources nor total power in ods, unable to calculate tau_e')
+                print("Not recalculating heating power, reusing ods['summary.global_quantities.power_steady.value']")
+                heating_power = ods['summary.global_quantities.power_steady.value'][time_index]
+            else:
+                ods.physics_summary_heating_power()
+                heating_power = ods['summary.global_quantities.power_steady.value'][time_index]
 
             # Calculate tau_e
             tau_e = abs(
                 56.2e-3
                 * (abs(ip) / 1e6) ** 0.93
                 * abs(bt) ** 0.15
                 * (ne_vol_avg / 1e19) ** 0.41
-                * (total_power / 1e6) ** -0.69
+                * (heating_power / 1e6) ** -0.69
                 * r_major ** 1.97
                 * kappa ** 0.78
                 * aspect ** -0.58
                 * isotope_factor ** 0.19
             )  # [s]
-            # print('kap', kappa, 'bt', bt, 'ip', ip, 'ne_vol', ne_vol_avg, 'paux', p_aux, 'aspect', aspect, 'isotope', isotope_factor, 'tau_e', tau_e)
+            if PRINTFLAG:
+                print('kap', kappa, 'bt', bt, 'ip', ip, 'ne_vol', ne_vol_avg, 'paux', p_aux, 'aspect', aspect, 'isotope', isotope_factor, 'tau_e', tau_e)
```

Comment:
ugly. remove PRINTFLAG  and use `printd()`
```suggestion
            for k in ['kappa', 'bt', 'ip', 'ne_vol_avg', 'p_aux', 'aspect', 'isotope_factor', 'tau_e']:
                printd(f'{k}: {eval(k)}', topic='summary_taue')
```

you can then see this output by setting `os.environ['OMAS_DEBUG_TOPIC']='summary_taue'`